### DISCUSSION TITLE
Add support for indicator host up and down

### DISCUSSION THREAD

**Discussion by opcache** [prometheus/pushgateway#543] [Q&A]
Add support for indicator host up and down

**Comment by beorn7** [comment]
Could you explain your proposal in more detail?

**Comment by opcache** [comment]
I use the pushgateway method to push alarm data, and the monitored host node is offline, how do I need an alarm notification？

**Comment by opcache** [comment]
I use the pushgateway method to push the alarm data, and I cannot see the up indicator from Prometheus, only the indicator of the pushgateway, as follows:
```
up{instance="127.0.0.1:9091", job="pushgateway", service="monitor"}
```
In pushgateway mode, other nodes use curl to send monitoring data to pushgateway. When pushgateway confiscates the monitoring data received from node curl, which indicator should be used for alarm notification?

**Comment by beorn7** [comment]
Fundamentally: The synthetic `up` metric is a byproduct of the pull-based metric collection. It cannot work with a push-based approach.

You can alert on pushers failing to push by checking on how long ago the last push happened.

**Reply by worker24h** [reply]
In k8s， when pod restart or teardown，but pushgateway is alived，however prometheus pulled the last valid data by pushgateway.   so I hope pushgateway don't cached data in TTL

**Reply by Jasmine456** [reply]
> 在k8s中，当pod重启或者teardown时，但是pushgateway是活跃的，而prometheus通过pushgateway拉取了最后的有效数据。所以我希望pushgateway不要在TTL内缓存数据

Can this problem be solved now
